src/renderer/components/cluster-manager/cluster-status.tsx

```diff
@@ -32,100 +32,92 @@ interface Dependencies {
 
 @observer
 class NonInjectedClusterStatus extends React.Component<ClusterStatusProps & Dependencies> {
-  @observable authOutput: KubeAuthUpdate[] = [];
-  @observable isReconnecting = false;
+  private readonly authOutputs = observable.map<ClusterId, KubeAuthUpdate[]>();
+  private readonly reconnecting = observable.set<ClusterId>();
 
-  constructor(props: ClusterStatusProps & Dependencies) {
-    super(props);
-    makeObservable(this);
-  }
-
-  get cluster(): Cluster {
-    return this.props.cluster;
-  }
+  private readonly authOutput = computed(() => this.authOutputs.get(this.clusterId) ?? []);
+  private readonly hasErrors = computed(() => this.authOutput.get().some(({ isError }) => isError));
+  private readonly isReconnecting = computed(() => this.reconnecting.has(this.clusterId));
+  private readonly clusterName = computed(() => {
+    const entity = catalogEntityRegistry.getById(this.clusterId);
+    const { cluster } = this.props;
 
-  @computed get entity() {
-    return catalogEntityRegistry.getById(this.cluster.id);
-  }
+    return entity?.getName() ?? cluster.name;
+  });
 
-  @computed get hasErrors(): boolean {
-    return this.authOutput.some(({ isError }) => isError);
+  private get clusterId() {
+    return this.props.cluster.id;
   }
 
   componentDidMount() {
     disposeOnUnmount(this, [
-      ipcRendererOn(`cluster:${this.cluster.id}:connection-update`, (evt, res: KubeAuthUpdate) => {
-        this.authOutput.push(res);
+      ipcRendererOn(`cluster:connection-update`, (evt, clusterId: ClusterId, update: KubeAuthUpdate) => {
```

Comment:
Explicit side-effect in touched (if not seen) code should be consolidated to a dependency

Reply:
May I ask you to please stop resolving conversation threads which are not resolved :)

Reply:
How is it not resolved? I have moved it into a dependency.